### PR TITLE
PHP 8 support

### LINKED ISSUE #14: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by tobias-trozowski**
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description
Updating to add PHP 8.0 support for #14 

In order to make this repository compatible, one has to follow these steps:

- [x] Modify composer.json to provide support for PHP 8.0 by adding the constraint ~8.0.0
- [x] Modify composer.json to drop support for PHP less than 7.3
- [x] Modify composer.json to implement phpunit 9.3 which supports PHP 7.3+
- [x] Modify .travis.yml to ignore platform requirements when installing composer dependencies (simply add --ignore-platform-reqs to COMPOSER_ARGS env variable)
- [x] Modify .travis.yml to add PHP 8.0 to the matrix (NOTE: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [x] Updated phpunit.xml.dist to fullfil schema

Closes #14

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,38 +12,28 @@ env:
...
+    - php: 7.4
       env:
         - DEPS=latest
-    - php: 7.3
+    - php: nightly
       env:
         - DEPS=lowest
-    - php: 7.3
+        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
+    - php: nightly
```
Comment (froschdesign) [inline .travis.yml]: At the moment version `8.0.1-dev` is used on Travis therefore it should be replaced with `8.0`.

See: https://travis-ci.com/github/laminas/laminas-i18n-resources/jobs/454999818#L211

**Comment by froschdesign** [comment]
@tobias-trozowski 
_I'm sorry for the late response._

The [XML configuration file for PHPUnit](https://github.com/laminas/laminas-i18n-resources/blob/2.7.x/phpunit.xml.dist) also needs an update:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
         bootstrap="vendor/autoload.php"
         colors="true">
    <testsuites>
        <testsuite name="laminas-i18n-resources Test Suite">
            <directory>./test</directory>
        </testsuite>
    </testsuites>
    <coverage>
        <include>
            <directory suffix=".php">./src</directory>
        </include>
    </coverage>
</phpunit>
```

Reference: https://phpunit.readthedocs.io/en/9.5/configuration.html#the-coverage-element

**Thanks in advance!**

**Comment by tobias-trozowski** [comment]
@froschdesign all done

**Review by froschdesign — APPROVED** [review]

**Review by Slamdunk — APPROVED** [review]

**Comment by froschdesign** [comment]
@tobias-trozowski 
Could you rebase against the branch 2.7.x? This will use the new CI workflow.
Thanks in advance! 👍

**Comment by froschdesign** [comment]
Fixed with #18

**Pull request closed without merging**